src/main/java/roomescape/service/ReservationService.java

```diff
@@ -0,0 +1,67 @@
+package roomescape.service;
+
+import org.springframework.stereotype.Service;
+import roomescape.domain.Reservation;
+import roomescape.domain.ReservationTime;
+import roomescape.dto.SaveReservationRequest;
+import roomescape.dto.SaveReservationTimeRequest;
+import roomescape.repository.ReservationRepository;
+import roomescape.repository.ReservationTimeRepository;
+
+import java.util.List;
+import java.util.NoSuchElementException;
+
+@Service
+public class ReservationService {
+    private final ReservationRepository reservationRepository;
+    private final ReservationTimeRepository reservationTimeRepository;
+
+    public ReservationService(
+            final ReservationRepository reservationRepository,
+            final ReservationTimeRepository reservationTimeRepository
+    ) {
+        this.reservationRepository = reservationRepository;
+        this.reservationTimeRepository = reservationTimeRepository;
+    }
+
+    public List<Reservation> getReservations() {
+        return reservationRepository.findAll();
+    }
+
+    public Reservation saveReservation(final SaveReservationRequest request) {
+        return reservationTimeRepository.findById(request.timeId())
+                .map(request::toReservation)
+                .map(reservationRepository::save)
+                .orElseThrow(() -> new NoSuchElementException("해당 id의 예약 시간이 존재하지 않습니다."));
```

Comment:
`orElseThrow`는 `findById`에 의해서 발생하는 예외인데요. 사실 메소드 체이닝으로 깔끔해보이긴하지만 "찾는것"과 "저장하는것"은 하나의 stream에 처리하기엔 너무 많은 기능을 차지하고 있어보여요. 

```suggestion
        ReservationTime reservationTime = reservationTimeRepository.findById(request.timeId())
                .orElseThrow(() -> new NoSuchElementException("해당 id의 예약 시간이 존재하지 않습니다."));

        return reservationRepository.save(request.toReservation(reservationTime));
```
stream 체이닝은 개인적으로 1개가 넘어가면 불필요한 가독성이 발생한다고 생각하는 편입니다.

Reply:
`찾는것과 저장하는것을 동시에`라는 구절이 와닿았습니다. 해당 코드는 제가 직접 작성하였기에 읽고 이해하는데 큰 무리가 없지만, 제 코드를 처음 읽는 개발자의 입장에서는 코드를 읽는데 조금 시간이 걸릴 수 있겠다는 생각이 드네요.

`orElseThrow`역시 `찾는것`에 대한 예외 처리지 `저장하는것`에 대한 예외처리가 아니니까요! 조언 감사합니다! 반영해보겠습니다!